python/cudf/reshape/general.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+import numpy as np
+import pandas as pd
+
+from cudf.dataframe import Series
+from cudf.dataframe import Buffer
+from cudf.dataframe import DataFrame
+from cudf.utils import cudautils
+from cudf.dataframe.categorical import CategoricalColumn
+
+
+def melt(frame, id_vars=None, value_vars=None, var_name='variable',
+         value_name='value'):
+    """Unpivots a DataFrame from wide format to long format,
+    optionally leaving identifier variables set.
+
+    Parameters
+    ----------
+    frame : DataFrame
+    id_vars : tuple, list, or ndarray, optional
+        Column(s) to use as identifier variables.
+        default: None
+    value_vars : tuple, list, or ndarray, optional
+        Column(s) to unpivot.
+        default: all columns that are not set as `id_vars`.
+    var_name : scalar
+        Name to use for the `variable` column.
+        default: frame.columns.name or 'variable'
+    value_name : str
+        Name to use for the `value` column.
+        default: 'value'
+
+    Returns
+    -------
+    molten : DataFrame
```

Comment:
Why does it say "molten" here?

Reply:
The past participle of _melt_ just like [_merged_](https://github.com/rapidsai/cudf/blob/branch-0.6/python/cudf/dataframe/dataframe.py#L948). But you're right, it should be [_melted_](https://brians.wsu.edu/2016/05/25/molten-melted/).

Reply:
done